### PR TITLE
Add bandwidth limiting support

### PR DESCRIPTION
**Pull request by CallMeFoxie**
**- What I did**
Added support for bandwidth limiting CNI plugins (it already exists within dockershim in kubelet). Default burst values have been copied from Kubelet (although those are `int` instead of `int64`, but I would argue that `int` is not enough on 10Gbit networks).

Please note that this  PR depends on merging https://github.com/cri-o/ocicni/pull/28 - after which this code needs to be revendored! Until then it will not compile.

**- How to verify it**
Create a pod with these annotations:
```
      annotations:
        kubernetes.io/egress-bandwidth: 20M
        kubernetes.io/ingress-bandwidth: 20M
```

and when you launch it there should be `qdisc` set:
```
# tc qdisc show
...
qdisc tbf 1: dev cali9d94ec92eeb root refcnt 2 rate 20Mbit burst 171798690b limit 62499b 
qdisc tbf 1: dev 76ae root refcnt 2 rate 20Mbit burst 171798690b limit 62499b 
...
```
**- Description for the changelog**
Added bandwidth CNI limiting support

### CODE REVIEW

**Comment by CallMeFoxie** [comment]
Builds are expected to fail until the OCICNI is merged and this MR is updated with the newer vendored package :)

**Comment by TomSweeneyRedHat** [comment]
@CallMeFoxie yeah, I noted that after I pressed the test button.  DOH!

File: server/utils.go [hunk 1]
```diff
@@ -99,12 +101,47 @@ func parseDNSOptions(servers, searches, options []string, path string) error {
 }
 
 func newPodNetwork(sb *sandbox.Sandbox) ocicni.PodNetwork {
+	var egress, ingress int64 = 0, 0
+
+	if val, ok := sb.Annotations()["kubernetes.io/egress-bandwidth"]; ok {
+		egressQ, err := resource.ParseQuantity(val)
+		if err != nil {
+			egress = 0
```
Comment (giuseppe) [inline server/utils.go]: shouldn't we fail here if we cannot parse the specified limit?

File: server/utils.go [hunk 2]
```diff
@@ -99,12 +101,47 @@ func parseDNSOptions(servers, searches, options []string, path string) error {
...
+		} else if iegress, isok := egressQ.AsInt64(); isok {
+			egress = iegress
+		}
+
+	}
+	if val, ok := sb.Annotations()["kubernetes.io/ingress-bandwidth"]; ok {
+		ingressQ, err := resource.ParseQuantity(val)
+		if err != nil {
+			ingress = 0
```
Comment (giuseppe) [inline server/utils.go]: same here

**Comment by umohnani8** [comment]
@CallMeFoxie can you please address the comments and test failures so we can get this moving along :)

**Comment by haircommander** [comment]
@umohnani8, @CallMeFoxie can't make movement until someone approves: https://github.com/cri-o/ocicni/pull/28

**Comment by haircommander** [comment]
hey @CallMeFoxie https://github.com/cri-o/ocicni/pull/28 is merged can you fetch the vendor and repush? :)

**Comment by haircommander** [comment]
LGTM

**Comment by haircommander** [comment]
LGTM @umohnani8 @mrunalp @giuseppe @rhatdan PTAL

**Pull request merged**